aws/macie_test.go

```diff
@@ -1,46 +1,61 @@
-// from our nuclear-wasteland account (where cloud-nuke tests are run), just so that we can nuke it again
-//
-// Macie is also regional, so for the purposes of cost-savings and lower admin overhead, we're initially only testing this
-// in the one hardcoded region - us-east-1
-//
-// The other reason we only test in us-east-1 is to avoid conflict with our Macie test in the CIS service catalog, which uses
-// these same two accounts for similar purposes, but in EU regions.
-// See: https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/blob/master/test/security/macie_test.go
-//func acceptTestInvite(t *testing.T, session *session.Session) {
-//	svc := macie2.New(session)
-//
-//	// Accept the "standing" invite from our other test account to become a Macie member account
-//	// This works because Macie invites don't expire or get deleted when you disassociate your member account following an invitation
-//	acceptInviteInput := &macie2.AcceptInvitationInput{
-//		AdministratorAccountId: aws.String("353720269506"),                     // sandbox
-//		InvitationId:           aws.String("18c0febb89142640f07ba497b19bac8e"), // "standing" test invite ID
-//	}
-//
-//	_, acceptInviteErr := svc.AcceptInvitation(acceptInviteInput)
-//	require.NoError(t, acceptInviteErr)
-//}
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/macie2"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/cloud-nuke/telemetry"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Macie tests are limited to testing the ability to find and disable basic Macie
+// features. The functionality of cloud-nuke disassociating/deleting members and
+// disassociating administrator accounts requires the use of multiple AWS accounts and the
+// ability to send and accept invitations within those accounts.
+
+func TestMacie(t *testing.T) {
+	telemetry.InitTelemetry("cloud-nuke", "", "")
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	require.NoError(t, err)
+	logging.Logger.Infof("Region: %s", region)
+
+	session, err := session.NewSession(&aws.Config{Region: aws.String(region)})
+	require.NoError(t, err)
+
+	svc := macie2.New(session)
+
+	// Check if Macie is enabled
+	_, err = svc.GetMacieSession(&macie2.GetMacieSessionInput{})
+
+	if err != nil {
+		// GetMacieSession throws an error if Macie is not enabled
+		if strings.Contains(err.Error(), "Macie is not enabled") {
+			logging.Logger.Infof("Macie not enabled.")
+			logging.Logger.Infof("Enabling Macie")
+			_, err := svc.EnableMacie(&macie2.EnableMacieInput{})
+			require.NoError(t, err)
+		} else {
+			require.NoError(t, err)
+		}
+	} else {
+		logging.Logger.Infof("Macie already enabled")
+	}
+
+	macieEnabled, err := getMacie(session, time.Now())
+	require.NoError(t, err)
+
+	logging.Logger.Infof("Nuking Macie")
+	nukeMacie(session, macieEnabled)
```

Comment:
I was wondering why error returned `nukeMacie()` is not checked

Reply:
ah, I just missed it, fixed!